main.js

```diff
@@ -1,163 +1,312 @@
+    });
+
+    let totalStock = 0;
+    for (let i = 0; i < stockArray.length; i++) {
+        totalStock = totalStock + stockArray[i];
+    }
+    return totalStock.toString();
+}
+
+//returns the result to index.html
+document.getElementById("stockCounter").innerHTML = totalStockCounter();
+
+//2a. array of all type names -------------------------------------------
+const typeNames = inventory.map((item) => {
+    return item.name;
+});
+
+//2b. array of all sold-out items (in a function that is called when the
+// sold-out button is clicked. The buttons are to be found at the Bonus exercise)
+function soldOut() {
+    const soldOut = inventory.filter((item) => {
+        return (item.originalStock - item.sold) === 0;
+    });
+    return document.getElementById("allProducts").innerHTML = allProducts(soldOut);
+}
+
+//2c. array of all items with AmbiLight (in a function that is called when the
+// hasAmbi button is clicked. The buttons are to be found at the Bonus exercise)
+function hasAmbi() {
+    const hasAmbi = inventory.filter((item) => {
+        return item.options.ambiLight === true;
+    });
+    document.getElementById("allProducts").innerHTML = allProducts(hasAmbi);
+}
+
+//2d. array of all items listed from lowest to highest price (in a function that is called
+//when low to high button is clicked. The buttons are to be found at the Bonus exercise)
+function lowestToHighestPrice() {
+    const lowestToHighestPrice = inventory.sort((a, b) => {
+        return a.price - b.price;
+    });
+    return document.getElementById("allProducts").innerHTML = allProducts(lowestToHighestPrice);
+}
+
+//3a. Target: Amount when all is sold-out ------------------------------
+function target() {
+    const targetPerItem = inventory.map((item) => {
+        return item.originalStock * item.price;
+    });
+
+    let targetCounter = 0;
+    for (let i = 0; i < targetPerItem.length; i++) {
+        targetCounter = targetCounter + targetPerItem[i];
+    }
+    return targetCounter.toString();
+}
+
+//returns the result to index.html
+document.getElementById("targetCounter").innerHTML = target();
+
+//3b. Counter: Current amount sold --------------------------------------
+function sold() {
+    const totalAmountSoldPerItem = inventory.map((item) => {
+        return (item.originalStock - item.sold) * item.price;
+    });
+
+    let counter = 0;
+    for (let i = 0; i < totalAmountSoldPerItem.length; i++) {
+        counter = counter + totalAmountSoldPerItem[i];
+    }
+    return counter.toString();
+}
+
+//returns the result to index.html
+document.getElementById("soldCounter").innerHTML = sold();
+
+//4. shows two type names ------------------------------------------------
+const typeNameOne = inventory[0].name;
+const typeNameTwo = inventory[1].name;
+
+//returns the result to index.html
+document.getElementById("inTheSpotLightsOne").innerHTML = typeNameOne;
+document.getElementById("inTheSpotLightsTwo").innerHTML = typeNameTwo;
```

Comment:
Deze twee had je mischien kunnen combineren tot 1 innerhtml-call, omdat ze in de html toch onder elkaar staan met niks er tussen